sale_stock_available_to_promise_release_cutoff/models/sale_order.py

```diff
@@ -0,0 +1,29 @@
+# Copyright 2021 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl)
+
+from odoo import api, models
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    @api.depends("order_line.availability_status", "picking_policy")
+    def _compute_display_expected_date_ok(self):
+        """Conditions to display the expected date on the sale_order report.
+
+        Displayed if:
+            - date is set (defined in sale_stock_available_to_promise_release)
+            - policy is 'when all products are ready' and everything is available
+            - policy is 'as soon as possible' and at least one line is available
+        """
+        super()._compute_display_expected_date_ok()
```

Comment:
move the check on original module to its own method so that you don't run on the whole recordset two times